api/storage_service.cc

```diff
@@ -446,12 +446,16 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
     });
 
     ss::describe_any_ring.set(r, [&ctx, &ss](std::unique_ptr<request> req) {
-        return make_ready_future<json::json_return_type>(stream_range_as_array(ss.local().describe_ring(""), token_range_endpoints_to_json));
+        return ss.local().describe_ring("").then([] (std::vector<dht::token_range_endpoints> endpoints) {
+            return json::json_return_type(stream_range_as_array(endpoints, token_range_endpoints_to_json));
+        }).finally([req = std::move(req)] {});
```

Comment:
Do we really need to keep req here?

Reply:
yeah, I was overly cautious.
We don't that anywhere else with an async callback.
I'll remove it.

Reply:
Fixed in v2